allennlp/nn/parallel/ddp_accelerator.py

```diff
@@ -27,7 +28,7 @@
     from allennlp.models import Model
 
 
-StateDictType = Union[Dict[str, torch.Tensor], OrderedDict[str, torch.Tensor]]
+StateDictType = Union[Dict[str, torch.Tensor], Mapping[str, torch.Tensor]]
```

Comment:
I had to change this to Mapping because with `OrderedDict` from collections I got `TypeError: 'type' object is not subscriptable `

Reply:
Hmm, have you tried quoting `OrderedDict`? Like
```suggestion
StateDictType = Union[Dict[str, torch.Tensor], "OrderedDict[str, torch.Tensor]"]
```

Reply:
Sorry - haven't got time to respond and test. @epwalsh are you going to take this over? Saw you self-assigned.

Reply:
Hey @ethanjyx, unfortunately we dropped support for 3.6 as of https://github.com/allenai/allennlp/pull/5400.

Reply:
@epwalsh thanks for letting me know that, that's helpful.

But you still might want to consider doing this PR, since OrderedDict from typing is deprecated since Python 3.9 and this code will break.

Reply:
Good point! Well if you'd like to finish it up I'm happy to reopen it then

Reply:
I'll finish it up in January. Pls reopen. Thanks

Reply:
@epwalsh looks like your suggested approach did not pass lint.. any ideas? not familiar with types within quotes